src/creatwth.cpp

```diff
@@ -1080,7 +1080,7 @@ BOOL WINAPI AllocExeHelper(_Out_ PDETOUR_EXE_HELPER *pHelper,
         cSize += (DWORD)cchDest + 1;
     }
 
-    Helper = (PDETOUR_EXE_HELPER) new NOTHROW BYTE[sizeof(DETOUR_EXE_HELPER) + cSize];
+    Helper = (PDETOUR_EXE_HELPER)DetourCreateObjectArray<BYTE>(sizeof(DETOUR_EXE_HELPER) + cSize);
```

Comment:
Why can't we just pass `PDETOUR_EXE_HELPER` to as the template parameter? 
```suggestion
    Helper = (PDETOUR_EXE_HELPER)DetourCreateObjectArray<PDETOUR_EXE_HELPER>(sizeof(DETOUR_EXE_HELPER) + cSize);
```

Reply:
Because DETOUR_EXE_HELPER is only the size of the header of the payload, plus cSize*sizeof(BYTE) is the entire size of the payload. If write it as you said, it just means that the memory space of the size of cSize*sizeof(PDETOUR_EXE_HELPER) is allocated. PDETOUR_EXE_HELPER is just a pointer type, so its size is only 4 or 8. So this is not in line with the original intention.

Reply:
Sure, read the code wrong. Thanks!